### ISSUE TITLE
[cls] Provide height & width per images to optimize CLS

### ISSUE THREAD

**Issue by AucT** [jetthoughts/jetthoughts.github.io#34]
## Acceptance Criteria:

All images on all pages have pre-filled width and height attributes, and page speed does not mention this issue.

![PageSpeed Insights 2024-09-11 10-52-04](https://github.com/user-attachments/assets/1c71cf17-e162-4cd5-88ff-ba50312a7deb)

## Details

For example you can check folder `content/blog/vertical-align-with-full-screen-across-tailwind-css-jetthoughts`

## Page bundles

As you've seen, I used a page bundle for every content. It's a more flexible and better approach, in my opinion.

So, you need to create a folder for the blog post page bundle.

For example, I converted https://jetthoughts.com/blog/vertical-align-with-full-screen-across-tailwind-css-jetthoughts/ to a page bundle.

The folder for the page bundle

`/content/blog/vertical-align-with-full-screen-across-tailwind-css-jetthoughts/`
You place your markdown inside the folder as `index.md` and place any images or other files associated with this post in this folder.

## Shortcodes

For images in blog post, I propose using my created shortcode.

Example of usage

```
![Image description](file_0.png)
```
file_0.png is an image inside the page bundle.

This will render the image and add width and height attributes.

You can also create other shortcodes to have your image resized by Hugo or add your CDN prefix if you like without the abomination you have.

## Fix post image open graph.

For displaying opengraph and twitter image meta tags in blog post add this

```
metatags:
  image: cover.jpg
```

cover.jpg is image from page bundle folder. It will render appropriate metatag

**Comment by AucT** [comment]
Image size will remove 1 CLS problem



![Image](https://github.com/user-attachments/assets/02d8ad03-6af6-41e0-b9ee-383534bf65fd)
![Image](https://github.com/user-attachments/assets/b022256d-2d37-4e17-a2ef-a3f3f23daf44)

**Comment by pftg** [comment]
@AucT will this issue close the problem with CLS #32 ?

**Comment by AucT** [comment]
@pftg 1 cls issue connected to image

**Comment by AucT** [comment]
@dgorodnichy 
updated example with image

After moving to page bundle,
use markdown for image

`![Image description](file_0.png)`

**Comment by pftg** [comment]
@AucT please convert this issue to PR

**Comment by pftg** [comment]
@AucT please post explicit steps which are required from @dgorodnichy or create separate issue for him

**Comment by pftg** [comment]
Blocked by #57

**Comment by pftg** [comment]
@AucT we merged #57 now you can finish, also please check the issue with small images https://github.com/orgs/jetthoughts/projects/6?pane=issue&itemId=78480853

**Comment by AucT** [comment]
@pftg css for blog is forcing rescale images. New html, css doesn't have such issues

**Comment by pftg** [comment]
@AucT It's not much about CSS, but our settings for responsiveness show that it uses container size instead of page width to detect which image to load, and the container is so narrow that it selects smaller (for mobile) versions. If you give images a whole page width, then yes, CSS can fix it. 

Elsewhere, what is left for this particular issue to be closed?

**Comment by pftg** [comment]
@AucT please fix the problem after rebase.
